### ISSUE TITLE
Flesh out the CLI

### ISSUE THREAD

**Issue by garyttierney** [soulsmods/fstools-rs#29]
The CLI could be a little more capable than it currently is. Here are some ideas for commands it could support:

```
fstools create [matbin|param|file|...]
fstools get [matbin|param|file|...] -o [json|csv|raw|...]
fstools unpack -o folder [--recursive] -- unpack BDT and the BNDs within
```

And potentially some others.

**Comment by GompDS** [comment]
I'm interesting in contributing on this issue, but I'm trying to understand exactly what these options entail. This is how I interpreted it, but I'm probably reading it wrong.

create: I assume takes one of those options (ie matbin, param, file, ect.) as an argument and creates a boilerplate file of that type, then writing to an output path (though that's not listed in the example so I'm not sure).

get: I think this would take a format option, followed by a file path? Then choose one the output types and it would create a file with, for example, .json at the end of the filename.

unpack: This one seems like it would unpack the game folder and output to a folder of your choice.

**Comment by garyttierney** [comment]
> create: I assume takes one of those options (ie matbin, param, file, ect.) as an argument and creates a boilerplate file of that type, then writing to an output path (though that's not listed in the example so I'm not sure).

Yeah, that's about right. We'd be creating one of those files from some intermediate representation passed as user input (e.g. the output of a `get` command might be MSB->JSON, `create` would go JSON->MSB -- the verbs we're using for these commands might not be the best tbh).

> get: I think this would take a format option, followed by a file path? Then choose one the output types and it would create a file with, for example, .json at the end of the filename.

Yep, that's right, this is about getting data into other formats for inspection and editing. Related to that is the `describe` command, which would print information about content without doing any conversions: <https://github.com/soulsmods/fstools-rs/blob/main/crates/cli/src/describe.rs#L9-L25>

> unpack: This one seems like it would unpack the game folder and output to a folder of your choice.

Yeah, this is basically recursive UXM. It's already implemented here: <https://github.com/soulsmods/fstools-rs/blob/main/crates/cli/src/extract.rs#L32>, but the CLI code is a bit ad-hoc and not that well put together right now.

**Comment by GompDS** [comment]
That clears things up a lot. Thanks! Maybe instead of 'create' and 'get', names like 'writebin' and 'writetxt' could be more clear.
